js/src/Figure.js

```diff
@@ -237,6 +239,8 @@ var Figure = widgets.DOMWidgetView.extend({
         this.listenTo(this.model, "change:title_style", this.title_style_updated, this);
         this.listenTo(this.model, "change:background_style", this.background_style_updated, this);
         this.listenTo(this.model, "change:layout", this.change_layout, this);
+        this.listenTo(this.model, "change:legend_style", this.legend_style_updated, this);
+        this.listenTo(this.model, "change:legend_text", this.legend_text_updated, this);
```

Comment:
The same set of listeners is added twice. You can remove either of them. If you do remove the ones in line 199-200, you need to make `legend_g` an object level variable.